### PR TITLE
CRM-20508 - Civicrm_Core - create field with non-ascii chars only

### PR DESCRIPTION
**Pull request by arthuralmeidap**
It's generated a random and unique name for custom fields when it's label field has only non-ascii chars when creating a new custom field.

---

 * [CRM-20508: Unable to use non-roman characters in Custom fields](https://issues.civicrm.org/jira/browse/CRM-20508)

### CODE REVIEW

**Comment by jitendrapurohit** [comment]
reviewing this PR.

**Comment by jitendrapurohit** [comment]
would be nice if check is done on the main function munge() instead of where it is called. Something like - 

    diff --git a/CRM/Utils/String.php b/CRM/Utils/String.php
    index 5fdeb91b75..790c472771 100644
    --- a/CRM/Utils/String.php
    +++ b/CRM/Utils/String.php
    @@ -84,6 +84,11 @@ class CRM_Utils_String {
         // CRM-11744
         $name = preg_replace('/[^a-zA-Z0-9]+/', $char, trim($name));
 
    +    //If there are no ascii characters present.
    +    if ($name == $char) {
    +      $name = self::createRandom($len, ALPHANUMERIC);
    +    }
    +
         if ($len) {
           // lets keep variable names short
           return substr($name, 0, $len);

Also I've added a test for this at https://github.com/civicrm/civicrm-core/pull/10335, which fails even after this PR as `$param['column_name']` is unchanged. It passes if change is done in the main function(munge()).

**Comment by arthuralmeidap** [comment]
Hi @jitendrapurohit !
I did the change and worked fine to me! I did the change outside de CRM_Utils_String::munge because I didn't want any side-effects for whom uses these function.

File: CRM/Utils/String.php [hunk 1]
```diff
@@ -84,6 +84,14 @@ public static function munge($name, $char = '_', $len = 63) {
     // CRM-11744
     $name = preg_replace('/[^a-zA-Z0-9]+/', $char, trim($name));
 
+      //var_dump($name);exit;
```
Comment (jitendrapurohit) [inline CRM/Utils/String.php]: Can you please remove this debug statement from the code ?

Comment (arthuralmeidap) [inline CRM/Utils/String.php]: I forgot that! Thanks!

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by eileenmcnaughton** [comment]
jenkins, add to whitelist

**Comment by eileenmcnaughton** [comment]
(kicked off jenkins test & tried to add you to the whitelist, it never works for me though)

**Comment by eileenmcnaughton** [comment]
@arthuralmeidap that was quick! Was just trying to add a comment explaining!

**Comment by eileenmcnaughton** [comment]
test this please

**Pull request merged**